### ISSUE TITLE
BeEquivalentTo doesn't always choose normal properties over explicitly implemented properties

### ISSUE THREAD

**Issue by MartinDembergerR9** [fluentassertions/fluentassertions#2754]
### Description

I use my own implementation of Exception which implements a interface with a Message-Property. This property is of another type.

### Reproduction Steps

```csharp
[Fact]
public void Test()
{
    var instance = new MyClass()
    {
        MyError = 42,
    };
    var other = new MyClass()
    {
        MyError = 42,
    };

    instance.Should().BeEquivalentTo(other);
}

public class MyClass : Exception, IMyInterface
{
    public int MyError { get; set; }
    int IMyInterface.Message => MyError;
}

public interface IMyInterface
{
    int Message { get; }
}
```

### Expected behavior

The test passes.

### Actual behavior

The test failes with the message:

```
Message: 
Expected property instance.MyNamespace.IMyInterface.Message to be System.String, but found System.Int32.

With configuration:
- Use declared types and members
- Compare enums by value
- Compare tuples by their properties
- Compare anonymous types by their properties
- Compare records by their members
- Include non-browsable members
- Match member by name (or throw)
- Be strict about the order of items in byte arrays
- Without automatic conversion.


  Stack Trace: 
XUnit2TestFramework.Throw(String message)
TestFrameworkProvider.Throw(String message)
CollectingAssertionStrategy.ThrowIfAny(IDictionary`2 context)
AssertionScope.Dispose()
EquivalencyValidator.AssertEquality(Comparands comparands, EquivalencyValidationContext context)
ObjectAssertions`2.BeEquivalentTo[TExpectation](TExpectation expectation, Func`2 config, String because, Object[] becauseArgs)
ObjectAssertions`2.BeEquivalentTo[TExpectation](TExpectation expectation, String because, Object[] becauseArgs)
MyTestClass.Test() line 426
RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

### Regression?

In 6.12.0 it worked by with 6.12.1 it is failing.

### Configuration

net8.0
FluentAssertions 6.12.1

### Are you willing to help with a pull-request?

No

**Comment by ITaluone** [comment]
The problem here is, that the explicit implemented interface member is named _exactly_ how the property of the base class: `Message`.

Quick fix would be to rename the interface member different than the member in `Exception`: e.g. `Message1` already does the trick, but I think this is not the desired way..

**Comment by MartinDembergerR9** [comment]
Thanks for the fast answer.

I didn't recognized `WithoutMatchingRules` but that's exactly what I need.

Is it a known bug or by design that `WithoutMatchingRules` returns void?
So I have to write the assertion like
```csharp
instance.Should().BeEquivalentTo(other, opt => 
    {
        opt.WithoutMatchingRules();
        return opt;
    }
);
```

I get this failure also on the `BeBinarySerializable`. I think `BeBinarySerializable` should also set `WithoutMatchingRules`.
For now I create my own implementation of this check.

**Comment by dennisdoomen** [comment]
> ```cs
> instance.Should().BeEquivalentTo(other, opt => opt.WithoutMatchingRules());
> ```

This removes the default rules that are used to map the properties from the expectation to the subject and essentially makes `BeEquivalentTo` a no-op operation. 

In this case, the problem is that the matching rule maps a normal property to an explicitly implemented property even though the subject-under-test also has a normal property. Normal properties should always get priority over explicitly implemented properties. 

Regardless of that, what you're doing feels pretty nasty to me. You're hiding the base-class `Message` property that all exceptions have by another with a different type. And next to that, you're not following normal .NET conventions for building exceptions.

**Comment by warappa** [comment]
I was also just hit by this after upgrading from 6.2.0 to 6.2.1.

The interesting thing (as in the original issue above) the properties do have **_different_ return types** (like in my code), but FluentAssertion's matching still tries to match them.

Maybe (also?) a return type check went missing between 6.2.0 and 6.2.1?

**Comment by dennisdoomen** [comment]
This is fixed in 6.12.2
